apps/showcase/src/components/showcase/design-token/design-token-pres.component.ts

```diff
@@ -21,33 +22,40 @@ import { DatePickerInputPresComponent } from '../../utilities';
   encapsulation: ViewEncapsulation.None,
   changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class DesignTokenPresComponent {
+export class DesignTokenPresComponent implements OnDestroy {
+  private readonly styleLoader = inject(StyleLazyLoader);
 
   /**
    * Form group
    */
-  public form: FormGroup<{ theme: FormControl<string | null> }>;
+  public form: FormGroup<{ theme: FormControl<string | null> }> = inject(FormBuilder).group({
+    theme: new FormControl<string | null>('')
+  });
 
-  constructor(fb: FormBuilder, styleLoader: StyleLazyLoader) {
-    this.form = fb.group({
-      theme: new FormControl<string | null>('')
-    });
+  private readonly subscription = new Subscription();
 
+  constructor() {
     let style: HTMLElement | null = null;
     const cleanUpStyle = () => {
       if (style?.parentNode) {
         style.parentNode.removeChild(style);
         style = null;
       }
     };
-    this.form.valueChanges.subscribe((value) => {
-      cleanUpStyle();
-      if (value.theme === 'dark') {
-        style = styleLoader.loadStyleFromURL({href: 'dark-theme.css'});
-      } else if (value.theme === 'horizon') {
-        style = styleLoader.loadStyleFromURL({href: 'horizon-theme.css'});
-      }
-    });
+    this.subscription.add(
+      this.form.valueChanges.subscribe((value) => {
```

Comment:
should we also use `takeUntilDestroy()` here to get rid of the subscriptions array ?

Reply:
I can be done in another PR I will open an issue :)

Reply:
https://github.com/AmadeusITGroup/otter/issues/1874